web/impact/impact/tests/test_mentor_program_office_hour_list_view.py

```diff
@@ -0,0 +1,164 @@
+        self._assert_nil_response_values(
+            {'mentor_name': NON_EXISTENT_NAME}
+        )
+
+    def test_filters_by_finalist_id(self):
+        finalist, finalist_office_hours = self._create_user_office_hours(
+            mentor=False
+        )
+        self._check_response_values(
+            {'finalist_id': finalist.id},
+            finalist_office_hours
+        )
+
+    def test_filters_by_finalist_name(self):
+        finalist, finalist_office_hours = self._create_user_office_hours(
+            mentor=False
+        )
+        self._check_response_values(
+            {'finalist_name': finalist.full_name()},
+            finalist_office_hours
+        )
+
+    def test_filters_by_partial_finalist_name(self):
+        finalist, finalist_office_hours = self._create_user_office_hours(
+            mentor=False
+        )
+        self._check_response_values(
+            {'finalist_name': finalist.first_name},
+            finalist_office_hours
+        )
+
+    def test_nonexistent_finalist_id_returns_nothing(self):
+        finalist, finalist_office_hours = self._create_user_office_hours(
+            mentor=False
+        )
+        self._assert_nil_response_values(
+            {'finalist_id': NON_EXISTENT_ID}
+        )
+
+    def test_nonexistent_finalist_name_returns_nothing(self):
+        finalist, finalist_office_hours = self._create_user_office_hours(
+            mentor=False
+        )
+        self._assert_nil_response_values(
+            {'finalist_name': NON_EXISTENT_NAME}
+        )
+
+    def _check_response_values(self, params, office_hours):
+        response = self._get_response_as_logged_in_user(params)
+        self.assertEqual(response.data["count"], USER_COUNT)
+        self.assertTrue(self._compare_ids(office_hours, response))
+
+    def _assert_nil_response_values(self, params):
+        response = self._get_response_as_logged_in_user(params)
+        self.assertEqual(response.data["count"], 0)
+        self.assertEqual(response.data["results"], [])
+
+    def _create_user_office_hours(self, mentor=True):
+        if mentor:
+            user = ExpertFactory()
+            office_hours = MentorProgramOfficeHourFactory.create_batch(
+                USER_COUNT, mentor=user)
+        else:
+            user = EntrepreneurFactory()
+            office_hours = MentorProgramOfficeHourFactory.create_batch(
+                USER_COUNT, finalist=user)
+        return user, office_hours
+
+    def _get_response_as_logged_in_user(self, params=None):
+        self.login(email=self.basic_user().email)
+        return self.client.get(self.url, params)
+
+    def _compare_ids(self, office_hours, response):
+        office_hour_ids = [
+            office_hour.id for office_hour in office_hours]
+        result_ids = [
+            result['id'] for result in response.data["results"]
+        ]
+        return all([id in result_ids for id in office_hour_ids])
```

Comment:
there's a world where you have 2 rather than 2 comprehensions here i.e.

```
    def _compare_ids(self, office_hours, response):
	        result_ids = [
	            result['id'] for result in response.data["results"]
	        ]
	        return all([office_hour.id in result_ids for office_hour in office_hours])
```